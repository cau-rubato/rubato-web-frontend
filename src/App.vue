<template>App</template>

<script>
export default {
  name: "app",
  data() {
    return {};
  },
};
</script>

<style>
</style>
