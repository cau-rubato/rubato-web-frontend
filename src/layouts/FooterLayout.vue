<template>
  <div class="footer bg-dark">
    <div class="container p-5">
      <div class="row">
        <div class="col-9">
          <p class="lead copyright">
            {{ copyright }}
          </p>
          <p class="lead text-light" style="display: inline">
            , ALL RIGHTS RESERVED
          </p>
          <p class="text-light address">{{ address }}</p>
          <p class="text-light email">Email : {{ email }}</p>
        </div>
        <div class="col-3" style="text-align: right">
          <h2 class="sns" v-for="(sns, i) in sns_list" :key="i">
            <a :href="sns.href"><i :class="sns.icon"></i></a>
          </h2>
        </div>
      </div>
    </div>
  </div>
</template>

<script>
export default {
  name: "FooterLayout",
  data() {
    return {
      copyright: "@RUBATO",
      address:
        "서울특별시 동작구 흑석로 84 중앙대학교 서울캠퍼스 학생회관 607호",
      email: "rubato-cau@gmail.com",
      sns_list: [
        {
          name: "facebook",
          icon: "bi bi-facebook text-light",
          href: "https://www.facebook.com/CAUrubato",
        },
        {
          name: "instagram",
          icon: "bi bi-instagram text-light",
          href: "https://www.instagram.com/rubato_cau1985/",
        },
        {
          name: "youtube",
          icon: "bi bi-youtube text-light",
          href: "https://www.youtube.com/channel/UCHwzULktVXIoA0HotXKRNlg",
        },
      ],
    };
  },
};
</script>

<style>
.copyright {
  color: #277da1;
  display: inline;
}

.sns {
  display: inline;
  margin-left: 20px;
}
</style>