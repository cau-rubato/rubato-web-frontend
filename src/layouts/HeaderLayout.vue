<template>
  <nav class="navbar navbar-expand-lg sticky-top navbar-dark bg-dark">
    <div class="container px-5">
      <a class="navbar-brand" href="/">Rubato</a>

      <button
        class="navbar-toggler"
        type="button"
        data-bs-toggle="collapse"
        data-bs-target="#navbarSupportedContent"
        aria-controls="navbarSupportedContent"
        aria-expanded="false"
        aria-label="Toggle navigation"
      >
        <span class="navbar-toggler-icon"></span>
      </button>

      <div class="collapse navbar-collapse" id="navbarSupportedContent">
        <ul class="navbar-nav me-auto mb-2 mb-lg-0">
          <NavItem
            v-for="(menu, i) in menus"
            :key="i"
            :menu="menu"
            class="nav-item"
          />
        </ul>

        <button class="btn btn-success disabled donate me-2" type="submit">
          후원하기
        </button>

        <button class="btn btn-outline-light portal disabled" type="submit">
          로그인
        </button>
      </div>
    </div>
  </nav>
</template>

<script>
import NavItem from "@/layouts/components/NavItem.vue";

export default {
  name: "HeaderLayout",
  data() {
    return {};
  },

  props: {
    menus: Object,
  },

  components: {
    NavItem,
  },
};
</script>

<style>
</style>