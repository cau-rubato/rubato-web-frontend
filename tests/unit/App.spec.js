import { shallowMount, mount } from '@vue/test-utils'
import App from '@/App.vue'
import HeaderLayout from '@/layouts/HeaderLayout.vue'
import FooterLayout from '@/layouts/FooterLayout.vue'


describe('App', () => {
    it('App exists', () => {
        const wrapper = shallowMount(App)
        expect(wrapper.exists()).toBeTruthy()
    });

    it('Layout Assembly Test', () => {
        const wrapper = mount(App);

        expect(wrapper.findComponent(HeaderLayout).exists()).toBeTruthy();
        expect(wrapper.findComponent(FooterLayout).exists()).toBeTruthy();
    });
});