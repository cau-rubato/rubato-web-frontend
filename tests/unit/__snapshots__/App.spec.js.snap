// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App exists 1`] = `" App"`;

exports[`App renders props.msg when passed 1`] = `" App"`;
